test/e2e/agent/agent_updates_test.go

```diff
@@ -76,6 +76,104 @@ var _ = Describe("VM Agent behavior during updates", func() {
 			logrus.Info("Device updated to new image 🎉")
 		})
+
+		It("Should update to v4 with embedded application", Label("updates", "rh-77667"), func() {
+			// Check the device status right after bootstrap
+			response := harness.GetDeviceWithStatusSystem(deviceId)
+			device := response.JSON200
+			Expect(device.Status.Summary.Status).To(Equal(v1alpha1.DeviceSummaryStatusType("Online")))
+			Expect(*device.Status.Summary.Info).To(Equal("Bootstrap complete"))
+			Expect(device.Status.Updated.Status).To(Equal(v1alpha1.DeviceUpdatedStatusType("Unknown")))
+			var newImageReference string
+
+			harness.UpdateDeviceWithRetries(deviceId, func(device *v1alpha1.Device) {
+				currentImage := device.Status.Os.Image
+				logrus.Infof("current image for %s is %s", deviceId, currentImage)
+				repo, _ := parseImageReference(currentImage)
+				newImageReference = repo + ":v4"
+				device.Spec.Os = &v1alpha1.DeviceOSSpec{Image: newImageReference}
+				logrus.Infof("updating %s to image %s", deviceId, device.Spec.Os.Image)
+			})
+			harness.WaitForDeviceContents(deviceId, "the device is applying update renderedVersion: 2",
+				func(device *v1alpha1.Device) bool {
+					return conditionExists(device, "Updating", "True", string(status.UpdateStateApplyingUpdate))
+				}, "1m")
+
+			Expect(device.Status.Summary.Status).To(Equal(v1alpha1.DeviceSummaryStatusType("Online")))
+
+			harness.WaitForDeviceContents(deviceId, "the device is rebooting",
+				func(device *v1alpha1.Device) bool {
+					return conditionExists(device, "Updating", "True", string(status.UpdateStateRebooting))
+				}, "2m")
+
+			Eventually(harness.GetDeviceWithStatusSummary, LONGTIMEOUT, POLLING).WithArguments(
+				deviceId).Should(Equal(v1alpha1.DeviceSummaryStatusType("Rebooting")))
+
+			harness.WaitForDeviceContents(deviceId, "status.Os.Image gets updated",
+				func(device *v1alpha1.Device) bool {
+					return device.Status.Os.Image == newImageReference &&
+						conditionExists(device, "Updating", "False", string(status.UpdateStateUpdated))
+				}, "2m")
+
+			Eventually(harness.GetDeviceWithStatusSummary, LONGTIMEOUT, POLLING).WithArguments(
+				deviceId).Should(Equal(v1alpha1.DeviceSummaryStatusType("Online")))
+
+			// TODO(hexfusion): we were expecting this update status not to be unknown at this point
+			// related to: https://issues.redhat.com/browse/EDM-679
+			// Expect(device.Status.Updated.Status).ToNot(Equal(v1alpha1.DeviceUpdatedStatusType("Unknown")))
+
+			logrus.Infof("Device updated to new image %s 🎉", "flightctl-device:v4")
+			logrus.Info("We expect containers with sleep infinity process to be present but not running")
+			stdout, err := harness.VM.RunSSH([]string{"sudo", "podman", "ps"}, nil)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(stdout.String()).To(ContainSubstring("sleep infinity"))
+
+			logrus.Info("We expect containers with sleep infinity process to be present but not running 👌")
```

Comment:
What do you mean by not running?

Reply:
The image has the application not correctly defined so it will fail at running, that what it means